packages/kestrel_core/src/kestrel/mapping/data_model.py

```diff
@@ -374,17 +422,21 @@ def translate_dataframe(df: DataFrame, to_native_nested_map: dict) -> DataFrame:
         except KeyError:
             _logger.debug("No mapping for %s", col)
             mapping = None
+        if isinstance(mapping, dict):
+            mapping = [mapping]
         if isinstance(mapping, list):
-            transformer_names = {
-                x.get("ocsf_value") for x in mapping if x.get("ocsf_value")
-            }
+            transformer_names = _get_transformers(mapping)
             if len(transformer_names) > 0:
                 if len(transformer_names) > 1:
                     raise NotImplementedError("Multiple to OCSF value transformers")
                 else:
                     transformer_name = transformer_names.pop()
-                    df[col] = run_transformer_on_series(
-                        transformer_name, df[col].dropna()
-                    )
+                    if isinstance(transformer_name, dict):
+                        # Not actually a named function; it's a literal value map
+                        df[col] = df[col].replace(transformer_name)
+                    else:
+                        df[col] = run_transformer_on_series(
```

Comment:
May need a guard not to call `None` if it is passed back from `_get_transformers()`, or just not return `None` elements as in previous comment.